hub/codec/tests/test_codecs.py

```diff
@@ -0,0 +1,92 @@
+import pytest
+import numpy as np
+
+from hub.codec import PngCodec, NumPy, Lz4, Zstd, JpegCodec
+
+
+@pytest.mark.parametrize("from_config", [False, True])
+def test_png_codec(from_config: bool) -> None:
+    codec = PngCodec()
+    if from_config:
+        config = codec.get_config()
+        codec = PngCodec.from_config(config)
+    arr = np.ones((1000, 1000, 3), dtype="uint8")
+    bytes_ = codec.encode(arr)
+    arr_ = codec.decode(bytes_)
+    assert (arr == arr_).all()
+
+
+@pytest.mark.parametrize("single_channel", [False, True])
+def test_png_codec_config(single_channel: bool) -> None:
+    codec = PngCodec(single_channel)
+    config = codec.get_config()
+    assert config["id"] == "png"
+    assert config["single_channel"] == single_channel
+
+
+@pytest.mark.parametrize("single_channel", [False, True])
+def test_png_codec_single_channel(single_channel: bool) -> None:
+    codec = PngCodec(single_channel)
+    if single_channel:
+        arr = np.ones((1000, 1000, 1), dtype="uint8")
+    else:
+        arr = np.ones((1000, 1000), dtype="uint8")
+    bytes_ = codec.encode(arr)
+    arr_ = codec.decode(bytes_)
+    assert (arr == arr_).all()
+
+
+@pytest.mark.parametrize("from_config", [False, True])
+def test_jpeg_codec(from_config: bool) -> None:
+    codec = JpegCodec()
+    if from_config:
+        config = codec.get_config()
+        codec = JpegCodec.from_config(config)
+    arr = np.ones((1000, 1000, 3), dtype="uint8")
```

Comment:
please use `pytest.mark.parametrize` for shapes & dtypes. [here is an example of what i mean](https://github.com/activeloopai/Hub/blob/32f2dc3367955d7b7959275792b770c824d098dc/hub/core/chunk_engine/tests/test_fixed_shapes.py#L33).